CONTRIBUTING.md

```diff
@@ -0,0 +1,14 @@
+# How to Contribute
+
+1. Fork the repository
+2. Make your changes in a local clone of your fork
+4. For a higher chance of CI passing the first time, consider running the following commands and fixing any errors or warnings (CI is currently pinned to a specific nightly version, so results may not match exactly):
+  1. `cargo fmt`
```

Comment:
This should be `cargo +nightly fmt --all`.

We also currently use `cargo clippy --all-targets --all-features -- -D warnings -A clippy::type_complexity -A clippy::manual-strip` in CI.

I'd love to simplify that if we can (maybe by adding the relevant config to a clippy config file?).

I'd also love to move off of nightly cargo fmt, but I don't want to lose the import collapsing feature :)